src/gwt/src/org/rstudio/studio/client/workbench/views/jobs/view/JobsList.java

```diff
@@ -39,13 +41,40 @@
 
    public JobsList()
    {
+      RStudioGinjector.INSTANCE.injectMembers(this);
+      
       jobs_ = new HashMap<String, JobItem>();
 
       initWidget(uiBinder.createAndBindUi(this));
+      
+      // register handler for hide completed jobs pref
+      uiPrefs_.hideCompletedJobs().addValueChangeHandler(hide ->
```

Comment:
I think this experience still needs some refinement. In particular if you have a bunch of launcher jobs that have recently completed and you don't want to see them any more, what should you do? 

The answer is currently "turn the preference off and then on again" which seems a little awkward. I think this suggests that perhaps we're thinking about the whole hide/showing thing wrong (maybe we need to sort instead of show/hide, for example). However I think we should take this change as-is and see how it feels before we speculate on something more complicated to build.

Reply:
Agree.